pkg/deployment/collector.go

```diff
@@ -148,12 +148,26 @@ func (c *Collector) Get() *appsv1.Deployment {
 
 // Services returns a list of services to be deployed along with the all-in-one deployment
 func (c *Collector) Services() []*v1.Service {
-	selector := c.selector()
 	return []*v1.Service{
-		service.NewCollectorService(c.jaeger, selector),
+		service.NewCollectorService(c.jaeger, c.labels()),
 	}
 }
 
-func (c *Collector) selector() map[string]string {
-	return map[string]string{"app": "jaeger", "jaeger": c.jaeger.Name, "jaeger-component": "collector"}
+func (c *Collector) labels() map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/name":       c.name(),
+		"app.kubernetes.io/instance":   c.jaeger.Name,
```

Comment:
Looks reasonable - only one I wasn't sure about was the `instance`, where the [docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels) describe it as "A unique name identifying the instance of an application" with an example of `wordpress-abcxzy`. But for our purpose it enables objects to be selected by various criteria.

Reply:
I'll apply the same change to the other objects then. Thanks!

Reply:
Don't forget we will also need to support the `app` label for a while - was thinking that should be same as `app.kubernetes.io/name` value?

Reply:
Do you have existing applications in mind, or is there a case of an application that still reads the `app` instead of `app.kubernetes.io`? If it's about maintaining backwards compatibility, we need to keep `app: jaeger`.

Reply:
I guess the two areas to consider is:
1) Istio - which needs the `app` label and `app:jaeger` should be fine
2) Kiali - raised by Jay? where unique `app` label seemed to be a problem - but possibly this is not a blocker

Reply:
If they are also moving to use `app.kubernetes.io` instead of `app`, then I'd keep `app: jaeger` for backwards compatibility.

Reply:
Unclear what the timeframe is for them changing, but yes probably the best approach.